### PR TITLE
chore: Remove theme editor and show-in-code

### PR DESCRIPTION
**Pull request by Artur-**
It is now in Copilot

### CODE REVIEW

File: vaadin-dev-server/package.json
Comment (sissbruecker) [inline vaadin-dev-server/package.json:1]: - `web-test-runner.config.mjs` should also be removed
- You're removing the`@open-wc/dev-server-hmr` dependency which is used in `web-dev-server.config.mjs`
- There's also a `patch-application.js` and related NPM script that is only useful if the dev server and HMR is used.
- Looks like non of the Vaadin components in the dev / non-dev dependencies are used anymore, except for @vaadin/overlay.

Comment (sissbruecker) [reply vaadin-dev-server/package.json:1]: These dependencies also seem to be unused now: `vanilla-colorful`, `sinon`, `@types/sinon`, `@types/chai`

Comment (sissbruecker) [reply vaadin-dev-server/package.json:1]: A left-over module from the theme editor: `src/main/frontend/vendor/@vaadin/overlay/src/vaadin-overlay-position-mixin.js`

Comment (Artur-) [reply vaadin-dev-server/package.json:1]: Updated. Leaving web-dev-server there for now

**Review by MarcinVaadin — APPROVED** [review]

**Pull request merged**